[ivanleekk] iP

ivanleekk opened the pull request:
### Ax

> "I dont know anything about coding" -Anonymous friend

Ax is a simple todo list bot that will save you precious brainwaves! 🤯
AND you dont need to know coding to use it

- text-based
- simple to learn
- **BLAZING QUICK**

To start, follow these simple steps

1. Download it from [here](https://github.com/ivanleekk/ip/archive/refs/tags/Level-9.zip)
2. double click the file
3. use (todo| deadline| event) to create new tasks
4. use (list) to run it

- [x] I have tried my best
- [x] I can do better!


If you would like to see the main class, here it is. It has all the other functions abstracted away into other packages.
```java
package ax;

import ax.commands.Parser;
import ax.display.Ui;
import ax.storage.Storage;

import java.util.Scanner;

public class Ax {

    /**
     * the main command for Ax chatbot
     * @param args
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Ui.greet();
        Storage.readSave();
        while (true) {
            boolean done = Parser.getInput(scanner);
            if (done) {
                break;
            }
        }
        Ui.bye();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ivanleekk committed:
Rename bot to Ax and added divider line

ivanleekk committed:
Added a new bye msg

ivanleekk committed:
remove old logo

ivanleekk committed:
added a echo thing where you repeat what you type

ivanleekk committed:
simplify if else

ivanleekk committed:
make the list work

ivanleekk committed:
use printf

ivanleekk committed:
seperate mark and unmark, also use 1 index

ivanleekk committed:
added 3 classes for the types of items in checklist

ivanleekk committed:
change expected txt

ivanleekk committed:
fixed bug where 3rd word on are not captured

ivanleekk committed:
added error handling

ivanleekk committed:
added the delete function

ivanleekk committed:
added the delete function

ivanleekk committed:
try to do testing

ivanleekk committed:
working automated test

ivanleekk committed:
add in javadocs

ivanleekk committed:
style check

ivanleekk committed:
fix throwing exception

ivanleekk committed:
can create an empty file now

ivanleekk committed:
can now save a file with text inside

ivanleekk committed:
can now read from file too

ivanleekk committed:
handle if dir does not exist

ivanleekk committed:
handle localdate and parsing

ivanleekk committed:
Moved everything into a different class

ivanleekk committed:
Moved everything into a packages

ivanleekk committed:
Merge branch 'master' into add-gradle-support

ivanleekk committed:
edit build.gradle such that it works with the new names

ivanleekk committed:
add junit test for task package

ivanleekk committed:
add junit test for listitem

ivanleekk committed:
Merge branch 'add-gradle-support'

ivanleekk committed:
add more javadocs

ivanleekk committed:
apply code style

ivanleekk committed:
add find function

ivanleekk committed:
Merge branch 'branch-Level-9'

ivanleekk committed:
Merge branch 'branch-A-CodingStandard'

ivanleekk committed:
Merge branch 'branch-A-JavaDoc'

ivanleekk committed:
Contains checkstyle in gradle and used it to fix all files

ivanleekk committed:
add javafx to gradle and fix test checkstyle

ivanleekk committed:
working prototype

ivanleekk committed:
moved functionality to gui and change duke image

NereusWB922 commented on src/main/java/ax/task/Events.java:
```diff
@@ -0,0 +1,30 @@
+        return (
+                "[E] " +
+                        super.toString() +
+                        String.format(" (from: %s to: %s)", this.start.toString(), this.end.toString())
```
Perhaps having line break before operator "+" and String.format arguments to improve readability? 
```suggestion
                "[E] " + super.toString()
                        + String.format(
                                " (from: %s to: %s)", this.start.toString(), this.end.toString())
```

NereusWB922 commented on src/main/java/ax/task/Deadlines.java:
```diff
@@ -0,0 +1,25 @@
+package ax.task;
+
+import java.time.LocalDate;
+
```
Maybe can have header comments for some of your public classes?

NereusWB922 commented on src/test/java/ax/task/DeadlinesTest.java:
```diff
@@ -0,0 +1,24 @@
+class DeadlinesTest {
+
+    @Test
+    void setDone() {
```
Perhaps can follow the naming convention for testing method to improve readability (`featureUnderTest_testScenario_expectedBehavior()`)?

NereusWB922 commented on src/main/java/ax/task/Deadlines.java:
```diff
@@ -0,0 +1,25 @@
+
+import java.time.LocalDate;
+
+public class Deadlines extends ListItem {
```
Perhaps having singular class name?

NereusWB922 reviewed (commented):
Looks good overall.

ivanleekk committed:
added fxml

ivanleekk committed:
Merge branch 'branch-gui'

ncduy0303 commented on src/main/java/ax/Ax.java:
```diff
@@ -0,0 +1,28 @@
+public class Ax {
+
+    /**
+     * the main command for Ax chatbot
```
Javadoc can be improved to follow the styling standard.

ncduy0303 commented on src/main/java/ax/commands/Parser.java:
```diff
@@ -0,0 +1,120 @@
+package ax.commands;
+
+import ax.display.Ui;
+import ax.task.*;
```
Consider removing wildcard import

ncduy0303 commented on src/main/java/ax/commands/Parser.java:
```diff
@@ -0,0 +1,120 @@
+     * @return true if the user wants to exit, false otherwise
+     */
+    @SuppressWarnings("unused")
+    public static boolean getInput(Scanner scanner) {
```
This method is quite long. Can consider splitting it with more smaller methods to handle each case.

ncduy0303 commented on src/main/java/ax/storage/Storage.java:
Consider avoiding the use of magic numbers. I think it will be quite difficult to change the parsing later on if there is some slight changes into the commands entered by the user.

ncduy0303 reviewed (commented):
Overall, the code is easy to understand and follow. Consider update some of the format to follow the recommended styling for the project.

ivanleekk committed:
added img to README.md

ivanleekk committed:
Merge branch 'branch-Level-10'

ivanleekk committed:
Add assertion to ListItem.java

ivanleekk committed:
Merge pull request #1 from ivanleekk/branch-A-Assertions

Add assertion to ListItem.java

ivanleekk committed:
Refactor getInputString in Parser.java

Remove unnecessary toString calls in Parser.java

ivanleekk committed:
Merge pull request #2 from ivanleekk/branch-A-CodeQuality

Improve Code Quality

ivanleekk committed:
Created GitHub Action to automate testing of code

ivanleekk committed:
Merge pull request #3 from ivanleekk/branch-A-CI

Created GitHub Action to automate testing of code

ivanleekk committed:
Implement Reminders

Add Reminders Interface

Add getDueDate to ListItem

So that the Ui package can check the due date without type checking if it is an Event or Deadline

ivanleekk committed:
Merge pull request #5 from ivanleekk/branch-B-Reminders

Implement Reminders

ivanleekk committed:
Make MainLayout and ScrollPane resizeable

Change by binding the width and height of scrollpane to mainlayout

ivanleekk committed:
Add spacing between image and text

Add rounded corners to image

ivanleekk committed:
Change width of userInput to prevent overlap

ivanleekk committed:
Merge pull request #6 from ivanleekk/branch-A-BetterGui

Branch a better gui

ivanleekk committed:
Update README.md

ivanleekk committed:
Update User Guide

ivanleekk committed:
Update User Guide

Fix checkstyle errors

ivanleekk committed:
Fix User Guide bugs

ivanleekk committed:
Remove emojis to ensure cross compatability

Update User Guide with new outputs

Update Ui.png with new outputs

ivanleekk committed:
Update Testcases

ivanleekk committed:
Auto stash before merge of "master" and "origin/master"

ivanleekk committed:
Merge branch 'master' into branch-remove-overhead

ivanleekk committed:
Fix code quality issues

ivanleekk committed:
Remove more overhead

ivanleekk committed:
Merge pull request #7 from ivanleekk/branch-remove-overhead

Remove overhead

ivanleekk committed:
Update build.gradle
